### PR TITLE
curl: Fix librtmp variant

### PR DESCRIPTION
**Pull request by tobbez**
Fix the `librtmp` variant by packaging its required dependency and adding it as
a (conditional) dependency of the curl package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/rtmpdump/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    """rtmpdump is a toolkit for RTMP streams."""
+
+    homepage = "https://rtmpdump.mplayerhq.hu/"
+    git = "https://git.ffmpeg.org/rtmpdump.git"
+
+    maintainers("tobbez")
+
+    license("GPL-2.0-or-later")
+
+    version("2021-02-19", commit="f1b83c10d8beb43fcc70a6e88cf4325499f25857")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rtmpdump/package.py:19]: They no longer make releases? I guess it's fine to use the last commit.

File: var/spack/repos/builtin/packages/rtmpdump/rtmpdump-fix-chunk-size.patch [hunk 2]
```diff
@@ -0,0 +1,48 @@
+https://git.alpinelinux.org/aports/commit/main/rtmpdump/fix-chunk-size.patch?id=bf39fb1177ee77eee6c214a7393cc0054958ce08
+https://git.alpinelinux.org/aports/commit/main/rtmpdump/fix-chunk-size.patch?id=69bc162319b12e9b6c6d3ea345dbf7c218753594
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rtmpdump/rtmpdump-fix-chunk-size.patch:2]: I guess it's fine, but out of curiosity: why are those ancient patches not upstreamed?

File: var/spack/repos/builtin/packages/rtmpdump/package.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    variant("tls", default="openssl", description="TLS backend", values=("gnutls", "openssl"))
+
+    depends_on("openssl", when="tls=openssl")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rtmpdump/package.py]: Maybe put an upperbound on `@:3`? It's using deprecated API, so won't work with an eventual openssl 4.

Comment (haampie) [reply var/spack/repos/builtin/packages/rtmpdump/package.py]:
```suggestion
    depends_on("openssl@:3", when="tls=openssl")
```

**Review by haampie — COMMENTED** [review]
Thanks! Looks good to me, add an upperbound on openssl?

File: var/spack/repos/builtin/packages/rtmpdump/package.py (hunk 1 above)
Comment (tobbez) [reply var/spack/repos/builtin/packages/rtmpdump/package.py:19]: Yes. Last release was in 2011.

At this point, others packaging rtmpdump seem to be using git snapshots.

File: var/spack/repos/builtin/packages/rtmpdump/rtmpdump-fix-chunk-size.patch (hunk 2 above)
Comment (tobbez) [reply var/spack/repos/builtin/packages/rtmpdump/rtmpdump-fix-chunk-size.patch:2]: I haven't investigated in detail, but I assume upstream is not responsive, considering these patches are carried by at least two distributions (Alpine and Gentoo).

**Comment by tobbez** [comment]
Upper bound added for openssl.

**Review by haampie — COMMENTED** [review]
Thanks, looks excellent!

**Review by haampie — APPROVED** [review]

**Pull request merged**